private/dependency_tree_parser.bzl

```diff
@@ -169,6 +170,24 @@ def _generate_imports(repository_ctx, dependencies, explicit_artifacts, neverlin
             elif packaging == "aar":
                 import_rule = "aar_import"
                 jar_versionless_target_labels.append(target_label)
+            elif packaging in ["so", "dylib", "dll"]:
+                import_rule = "java_library"
+                is_dylib = True
+                jar_versionless_target_labels.append(target_label)
+                dylib = simple_coord.split(":")[-1] + "." + packaging
+                all_imports.append(
+                    """
+genrule(
+    name = "{dylib}_extension",
+    srcs = ["@{repository}//file"],
+    outs = ["{dylib}"],
+    cmd = "cp $< $@",
```

Comment:
these copy rules exist to bring the artifacts into the @maven// top-level, which lets the user have a single `java.library.path` flag